unstract/core/src/unstract/core/pubsub_helper.py

```diff
@@ -93,12 +90,53 @@ def log_prompt(
             "message": message,
         }
 
+    @classmethod
+    def _get_task_message(
+        cls, user_session_id: str, event: str, message: Any
+    ) -> dict[str, Any]:
+
+        task_kwargs = {
+            LogEventArgument.EVENT: event,
+            LogEventArgument.MESSAGE: message,
+            LogEventArgument.USER_SESSION_ID: user_session_id,
+        }
+        task_message = {
+            "args": [],
+            "kwargs": task_kwargs,
+            "retries": 0,
+            "utc": True,
+        }
+        return task_message
+
+    @classmethod
+    def _get_task_header(cls, task_name: str) -> dict[str, Any]:
+        return {
+            "task": task_name,
+        }
+
     @classmethod
     def publish(cls, channel_id: str, payload: dict[str, Any]) -> bool:
-        channel = f"logs:{channel_id}"
+        """Publish a message to the queue."""
         try:
-            cls.r.publish(channel, json.dumps(payload))
+            with cls.kombu_conn.Producer(serializer="json") as producer:
+                event = f"logs:{channel_id}"
+                task_message = cls._get_task_message(
+                    user_session_id=channel_id,
+                    event=event,
+                    message=payload,
+                )
+                headers = cls._get_task_header(LogProcessingTask.TASK_NAME)
+                # Publish the message to the queue
+                producer.publish(
+                    body=task_message,
+                    exchange="",
+                    headers=headers,
+                    routing_key=LogProcessingTask.QUEUE_NAME,
+                    compression=None,
+                    retry=True,
+                )
+                logging.info(f"Published '{channel_id}' <= {payload}")
```

Comment:
NIT: This could be a debug log if its too noisy

Reply:
updated